mono/utils/mono-dl-posix.c

```diff
@@ -47,7 +47,11 @@ mono_dl_get_so_suffixes (void)
 int
 mono_dl_get_executable_path (char *buf, int buflen)
 {
+#if defined(HAVE_READLINK)
 	return readlink ("/proc/self/exe", buf, buflen - 1);
+#else
+	return -1;
```

Comment:
Maybe set errno to ENOSYS or such? Looks like that wouldn't really do anything.
Or raise `#ifdef` out to caller? Seems reasonable, if obsessing over size.
Or have `#ifdef HAVE_MONO_DL_GET_EXECUTABLE_PATH` to abstract it? To let the caller decide what to do, if the function will always fail.